### ISSUE TITLE
Webpack 5 Compatibility

### ISSUE THREAD

**Issue by felixhao28** [cascornelissen/event-hooks-webpack-plugin#9]
The latest version of webpack is 5.2.0

**Comment by cascornelissen** [comment]
Thanks for creating an issue, this is on my to-do list but if anyone wants to submit a PR to support both Webpack 4 and 5 in the meantime that would be awesome as well ✌🏼

**Comment by felixhao28** [comment]
It already works out of the blue with Webpack 5 after ignoring warnings/errors on peerDependencies requiring Webpack 4.x. Except that the event "done" becomes "afterDone" for some reason.

**Comment by cascornelissen** [comment]
As you've probably found out by now, there's no list being kept in this plugin so new hooks should be supported automatically. A new format (see the [`processAssets` hook](https://webpack.js.org/api/compilation-hooks/#processassets) for example) where a hooks can have different stages does not seem to be supported so that's something that still needs to be added.

**Comment by felixhao28** [comment]
Is it possible to specify different callbacks for different stages of the same event?

**Comment by cascornelissen** [comment]
I'm pretty sure it is, yes, otherwise what would the use of stages be? But it's worth testing that I guess.

**Comment by felixhao28** [comment]
I mean, with the way you are writing in the example, how do you specify different callbacks for different stages? Like this?

```js
const EventHooksPlugin = require('event-hooks-webpack-plugin');

module.exports = {
    plugins: [
        new EventHooksPlugin({
            processAssets: {
                options: {
                    stage: Compilation.PROCESS_ASSETS_STAGE_ADDITIONS // Not sure where Compilation would be coming from in this case
                },
                callback: () => {
                    // ...
                }
            },
            processAssets2: {
                options: {
                    stage: Compilation.PROCESS_ASSETS_STAGE_DEV_TOOLING
                },
                callback: () => {
                    // ...
                }
            },

        })
    ]
};
```

**Comment by cascornelissen** [comment]
Ah, gotcha, the key should always match an existing Webpack hook so the best way to support multiple different callbacks for the same hooks would be arrays. We should probably add support for array values in general.

```es6
module.exports = {
    plugins: [
        new EventHooksPlugin({
            processAssets: [{
                options: {
                    stage: Compilation.PROCESS_ASSETS_STAGE_ADDITIONS
                },
                callback: () => {
                    // ...
                }
            }, {
                options: {
                    stage: Compilation.PROCESS_ASSETS_STAGE_DEV_TOOLING
                },
                callback: () => {
                    // ...
                }
            }]
        })
    ]
};
```

This is of course a lot more work and I can image you don't have the time for it. If you want me to, I can set this up later today as well.

**Comment by cascornelissen** [comment]
Whoops, this got closed automatically because of the commits in the PR. Webpack 5 should be supported now but we still have a couple steps to go:

- [ ] Adjust test setup to allow testing with both Webpack 4 and 5
- [ ] Improve support for hooks with stages

**Comment by felixhao28** [comment]
I don't particularly need the support for stages callbacks right now. So take your time.

**Comment by cascornelissen** [comment]
I have just published version `2.2.0` under the `next` dist-tag on NPM. Could you (and anyone else following this ticket) test the release out within your projects and report back? If everything goes well I'll release the same version with the `latest` dist-tag.

```
npm install event-hooks-webpack-plugin@next --save-dev
```

**Comment by felixhao28** [comment]
`2.2.0` works like a charm.

**Comment by cascornelissen** [comment]
Awesome, I've published [`2.2.0`](https://github.com/cascornelissen/event-hooks-webpack-plugin/releases/tag/v2.2.0) with the `latest` dist-tag on NPM. 🚀 Thanks for your contributions and help with testing ✌🏼
